codice/alliance
# MPEG-TS Streaming Fixes

## Do not squash on merge

#### What does this PR do?
- Fix MPEG-TS UDP streaming
  - The stream startup code had been erroneously replaced with the stream shutdown code.
  - The videographer user needs the 'guest' role to create metacards.
- Update location and frame center updaters to handle GeometryCollections
- Fix thread interrupt handling
- Fix and re-enable VideoTest
- Delete unused metacard updaters

#### Who is reviewing it?
@sgalla 
@aj-brooks 

#### Choose 2 committers to review/merge the PR.
@blen-desta 
@glenhein 

#### How should this be tested?
1. Install Alliance with the video app.
2. Configure a video stream in the Admin UI. The default settings will work fine, but you can change the title or filename template if you want.
3. Stream an MPEG-TS video to the address specified in the stream configuration from the previous step (the default is `127.0.0.1:50000`). Two sample videos are available here: https://samples.ffmpeg.org/MPEG2/mpegts-klv/
a. Using [FFmpeg](https://ffmpeg.org/): `ffmpeg -re -i Night_Flight_IR.mpg -map 0 -c copy -f mpegts udp://127.0.0.1:50000`
b. Using [TSduck](https://tsduck.io/): `tsp -I file Night_Flight_IR.mpg -P regulate -O ip 127.0.0.1:50000`
4. Wait for the video to finish streaming.
5. Open Intrigue and search for the stream title (for the parent metacard) and the filename template (for the chunks). Verify the parent metacard has a location and/or a frame center geometry. Verify the chunk metacards also have locations and/or frame center geometries.

#### Any background context you want to provide?
MPEG-TS UDP streaming has been broken since 1.4.0 due to [this change](https://github.com/codice/alliance/commit/b2200d94aaefd0239b3c1f2563b0be50c043066e#diff-2f6e154f5c64301bbb1ca52d100d4aefR432)

#### Checklist:
